pkg/services/grpcserver/context/context.go

```diff
@@ -0,0 +1,25 @@
+package grpccontext
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/user"
+)
+
+type grpcContextKey struct{}
+
+type GRPCServerContext struct {
+	SignedInUser *user.SignedInUser
+	Tracer       tracing.Tracer
+	Logger       log.Logger
+}
+
+func FromContext(ctx context.Context) *GRPCServerContext {
```

Comment:
just wondering - would we move this file to `pkg/services/user` at some point? so that we can set/retrieve the user in/from the context in all code paths (http, cli, grpc, ..)

Reply:
i should have mentioned this in the PR description. there's `pkg/services/contexthandler/contexthandler.go` middleware which uses `models.ReqContext` but that seems pretty specific to HTTP requests. it was unclear what types of undefined behavior would result if we used that without providing a `web.Contex`, so it seemed safer to add something specific for gRPC requests.

i'm not convinced that this is a good approach since i would prefer to not maintain separate code for a very similar use case. we could also try reusing the HTTP context handler and fix any issues as they pop up. thoughts?

Reply:
i'd say in an ideal scenario the http/grpc middlewares just extract the data we want from their respective technology-specific objects, and then we pass that data to a common function - 

```
httpAuthMiddleware(webContext) {
   genericAuthMiddleware(extractWebHttpAuthStuff(webContext))
}

grpcAuthMiddleware(grpcContext) {
   genericAuthMiddleware(extractGrpcAuthStuff(grpcContext))
}

```

but I am haven't look at the code - I am not sure how difficult would it be